### PR TITLE
Updated python bindings

### PR DESCRIPTION
**Pull request by padupr**
Updated python bindings and added xml writer

Closes se-passau/VaRa#667

### CODE REVIEW

File: bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "pybind_common.h"
+#include "vara/Feature/FeatureModelWriter.h"
+
```
Comment (vulder) [inline bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp]:
```suggestion
#include "vara/Feature/FeatureModelWriter.h"
#include "pybind_common.h"

```

File: bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#include <iostream>
+
+namespace vf = vara::feature;
+namespace py = pybind11;
+
+void init_xml_writer(py::module &M) {
+  py::class_<vf::FeatureModelXmlWriter>(M, "FeatureModelXmlWriter")
+      .def(py::init<const vf::FeatureModel &>())
+      .def(
+          "dumps_feature_model",
```
Comment (vulder) [inline bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp]: Why did you choose a different name here for the python side?

File: include/vara/Feature/Feature.h [hunk 3]
```diff
@@ -159,6 +159,9 @@ class Feature {
   [[nodiscard]] FeatureSourceRange *getFeatureSourceRange() {
     return Source.has_value() ? &Source.value() : nullptr;
   }
+  void setFeatureSourceRange(FeatureSourceRange &Fsr) {
```
Comment (vulder) [inline include/vara/Feature/Feature.h]:
```suggestion
  void setFeatureSourceRange(FeatureSourceRange FeatureSR) {
```
If your going to move the object afterwards anyhow taking it as a value is prefered.

File: include/vara/Feature/Feature.h [hunk 4]
```diff
@@ -159,6 +159,9 @@ class Feature {
...
+    Source = std::move(Fsr);
```
Comment (vulder) [inline include/vara/Feature/Feature.h]:
```suggestion
    Source = std::move(FeatureSR);
```

**Review by vulder — APPROVED** [review]
A few minor things but otherwise this looks good :)

File: bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp (hunk 2 above)
Comment (padupr) [reply bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp]: Python does not allow function overloading, so I used the naming of python's json lib to disambiguate the functions.

Comment (vulder) [reply bindings/python/vara-feature/pybind_FeatureModelXmlWriter.cpp]: Yes, that's you use `dump_feature_model` vs `dumps_feature_model` but why did you change write to dump?
`dump_feature_model` could be `write_feature_model_to`
`dumps_feature_model` could be `get_feature_model_as_string` or `convert_feature_model_to_string`

**Pull request merged**